src/formatters/utils/sortResults.ts

```diff
@@ -1,13 +1,54 @@
 import { IRuleResult } from '../../types';
 
-export const sortResults = (results: IRuleResult[]) => {
-  return [...results].sort((resultA, resultB) => {
-    const diff = resultA.range.start.line - resultB.range.start.line;
+const safeCastToString = (code: string | number | undefined): string => {
+  if (code === undefined) {
```

Comment:
@P0lip Are there any use case where code is either `undefined` or a `number`? Wouldn't that be the case, as `5.0` is around the corner, could we make this a non nullable string?

Reply:
AFAIK our own code includes code in each diagnostic result, but that doesn't necessarily need to be the case when it comes to end-user code. 
Moreover, if we wanted to make it required, we'd need to make the change over here https://github.com/stoplightio/types/blob/master/src/diagnostics.ts#L63, as that's is the interface `SpectralDiagnostic` inherits from, or, alternatively, we could make `code` required over here https://github.com/stoplightio/spectral/blob/develop/src/types/spectral.ts#L42#L44.
Either way, `code` is rather an additional metadata, not something really required, so I'd keep it optional.

Reply:
and yeah, we should not be throwing either.

Reply:
> and yeah, we should not be throwing either.

Makes sense. Pushed another proposal.